src/main/java/command/DeadlineCommand.java

```diff
@@ -0,0 +1,47 @@
+package command;
+
+import duke.Storage;
+import tasks.TaskList;
+import duke.Ui;
+import exceptions.DukeException;
+import tasks.Deadline;
+import tasks.Task;
+
+
+/**
+ * this class handles when a deadline command is called and executed
+ */
+public class DeadlineCommand extends Command {
+
+    private String input;
+
+    /**
+     * constructor for tasks.Deadline command.Command
+     *
+     * @param input
+     */
+    public DeadlineCommand(String input) {
+        super();
+        this.input = input;
+    }
+
+    /***
+     * splits the input and adds a deadline task into tasks.TaskList
+     * if duplicate detected, no changes made to taskList or storage
+     * @param taskList
+     * @param ui
+     * @param storage
+     * @throws DukeException
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
+        input = input.substring(9);
```

Comment:
Could you put this logic in the parser instead? The abstraction seems to make more sense there.